internal/xcrd/crd.go

```diff
@@ -156,9 +167,20 @@ func ForCompositeResourceClaim(xrd *v1.CompositeResourceDefinition) (*extv1.Cust
 			specProps.Properties[k] = v
 		}
 		crd.Spec.Versions[i].Schema.OpenAPIV3Schema.Properties["spec"] = specProps
+
+		statusP, statusRequired, err := getProps("status", vr.Schema)
+		if err != nil {
+			return nil, errors.Wrap(err, errGetSpecProps)
```

Comment:
Should we replace `errGetSpecProps` with something that covers status (or arbitrary fields)?